### PR TITLE
Cosmos: add basic support for nested collections of primitive types

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

File: src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs [hunk 1]
```diff
@@ -79,7 +77,9 @@ public CosmosTypeMappingSource(TypeMappingSourceDependencies dependencies)
 
             if (clrType.IsArray)
             {
-                var elementMapping = FindPrimitiveMapping(new TypeMappingInfo(elementType));
+                var elementMappingInfo = new TypeMappingInfo(elementType);
+                var elementMapping = FindPrimitiveMapping(elementMappingInfo)
```
Comment (roji) [inline src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs:81]: Any reason to not just call FindMapping here? That also seems important in order to provide support for collections over plugin-provided types (base.FindMapping)

File: src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs [hunk 2]
```diff
@@ -93,11 +93,12 @@ public CosmosTypeMappingSource(TypeMappingSourceDependencies dependencies)
...
                 {
-                    var elementMapping = FindPrimitiveMapping(new TypeMappingInfo(elementType));
+                    var elementMappingInfo = new TypeMappingInfo(elementType);
+                    var elementMapping = FindPrimitiveMapping(elementMappingInfo)
+                        ?? FindCollectionMapping(elementMappingInfo);
                     return elementMapping == null
                         ? null
-                        : new CosmosTypeMapping(clrType,
-                            CreateListComparer(elementMapping, elementType, clrType, genericTypeDefinition == typeof(IReadOnlyList<>)));
+                        : new CosmosTypeMapping(clrType, CreateListComparer(elementMapping, elementType, clrType));
```
Comment (roji) [inline src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs:101]: Seems like readOnly is never true now, or am I missing something?

File: test/EFCore.Cosmos.FunctionalTests/ValueConvertersEndToEndCosmosTest.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 #nullable enable
 
-namespace Microsoft.EntityFrameworkCore
+namespace Microsoft.EntityFrameworkCore.Cosmos
```
Comment (roji) [inline test/EFCore.Cosmos.FunctionalTests/ValueConvertersEndToEndCosmosTest.cs:8]: FYI ValueConvertersEndToEndSqlServerTest is in Microsoft.EntityFrameworkCore, we should clean up at some point

**Review by roji — APPROVED** [review]
Thanks @AndriySvyryd, see comments

File: src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs:81]: This is to make sure that only primitive types can be at the bottom

File: src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs (hunk 2 above)
Comment (AndriySvyryd) [reply src/EFCore.Cosmos/Storage/Internal/CosmosTypeMappingSource.cs:101]: Correct. This is safer, as now the type could be `IReadOnlyList<List<int>>`, which shouldn't be treated as readonly.

**Pull request merged**